pkg/storage/storage_types.go

```diff
@@ -0,0 +1,40 @@
+package storage
+
+import (
+	"github.com/pbinitiative/zenbpm/pkg/bpmn/model/bpmn20"
+	"time"
+)
+
+type ProcessDefinition interface {
+	// BpmnProcessId the ID as defined in the BPMN file
+	BpmnProcessId() string
+
+	// Version of the process, default=1, incremented, when another process with the same ID is loaded
+	Version() int32
+
+	// ProcessKey the engines key for this given BpmnProcessId with Version
+	ProcessKey() int64
```

Comment:
It would be nice to switch to snowflakeId as an id as soon as possible (idk if it is feasible in this PR)